### PR TITLE
Support for minimumRancherVersion property on templates and added a filter minimumRancherVersion_lte

### PR DESCRIPTION
**Pull request by prachidamle**
Added support for property 'minimumRancherVersion' on the templates .catalog section
Added a filter on the api:  /v1-catalog/templates?minimumRancherVersion_lte=<version>

Fixes https://github.com/rancher/rancher/issues/2489

### CODE REVIEW

**Comment by prachidamle** [comment]
@ibuildthecloud @vincent99 please review and merge this to support the minimumRancherVersion

@cloudnautique .catalog needs to contain minimumRancherVersion property for all template versions after this change is merged

**Comment by prachidamle** [comment]
@vincent99 @ibuildthecloud made the changes to have 'minimumRancherVersion' property as optional. So if a template version does not have this property, it will be listed always irrespective of the filter passed in.

**Comment by prachidamle** [comment]
@cloudnautique the property in rancher_compose.yml should be 'minimum_rancher_version'

File: service/handlers.go [hunk 1]
```diff
@@ -3,22 +3,45 @@ package service
...
+
+	//read the catalog
 	resp := model.TemplateCollection{}
 	for _, value := range manager.Catalog {
+
+		if rancherVersion != "" {
+			var err error
+			value.VersionLinks, err = filterByMinimumRancherVersion(rancherVersion, &value)
+			if err != nil {
+				//cannot apply the filter, return empty set
```
Comment (alena1108) [inline service/handlers.go]: would to make sense to log the error here?

File: service/handlers.go [hunk 2]
```diff
@@ -3,22 +3,45 @@ package service
...
+				break
+			}
+		}
+
+		//if no versions are present then just skip the template
+		if len(value.VersionLinks) == 0 {
```
Comment (alena1108) [inline service/handlers.go]: Should we skip the template is no version is present? I would expect to return it instead if the template creator didn't put the minimum version in.

Comment (prachidamle) [inline service/handlers.go]: Here, in case this means that if a template has no subversions (like a folder 0, 1,2 etc) at all under it, then that template is skipped. This is not related to skipping the template is there is no minimum rancher version

File: service/handlers.go (hunk 1 above)
Comment (prachidamle) [inline service/handlers.go]: error is already logged by the func filterByMinimumRancherVersion()

**Comment by alena1108** [comment]
LGTM

**Comment by cloudnautique** [comment]
LGTM
